### PR TITLE
Wanl/more asp e2e test

### PR DESCRIPTION
**Pull request by realwanpengli**

### CODE REVIEW

File: test/Microsoft.Azure.SignalR.AspNet.Tests/E2ETest/TestHub.cs [hunk 1]
```diff
@@ -1,15 +1,96 @@
...
+using System.Web.UI.WebControls;
 using Microsoft.AspNet.SignalR;
+using Serilog;
+using Serilog.Core;
 
 namespace Microsoft.Azure.SignalR.AspNet.Tests
 {
     public class TestHub : Hub
     {
+        private static ConcurrentDictionary<string, bool> _connectedConnections = new ConcurrentDictionary<string, bool>();
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.AspNet.Tests/E2ETest/TestHub.cs]: >static [](start = 16, length = 6)

why `static`?

File: test/Microsoft.Azure.SignalR.AspNet.Tests/E2ETest/TestServer.cs [hunk 2]
```diff
@@ -2,27 +2,59 @@
...
+        protected override Task StartCoreAsync(string serverUrl, ITestOutputHelper output)
+        {
+            TestHub.ClearConnectedConnectionAndUser();
+
+            _loggerFactory = new LoggerFactory().AddXunit(output);
+
+            _webApp = WebApp.Start(new StartOptions(serverUrl), app =>
+            {
+                var hubConfiguration = Utility.GetActualHubConfig(_loggerFactory);
+                hubConfiguration.Resolver.Register(typeof(IUserIdProvider), () => Activator.CreateInstance(typeof(UserIdProvider)));
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.AspNet.Tests/E2ETest/TestServer.cs]: >CreateInstance [](start = 92, length = 14)

singleton is enough

File: test/Microsoft.Azure.SignalR.AspNet.Tests/E2ETest/TestServer.cs [hunk 3]
```diff
@@ -2,27 +2,59 @@
...
+                GlobalHost.TraceManager.Switch.Level = SourceLevels.Information;
+            });
             return Task.CompletedTask;
         }
 
         public override Task StopAsync()
         {
-            _webApp.Dispose();
+            _webApp?.Dispose();
+            _loggerFactory.Dispose();
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.AspNet.Tests/E2ETest/TestServer.cs]: >Dispose [](start = 27, length = 7)

`?.` as `_loggerFactory` is not inited until `StartCoreAsync`

File: test/Microsoft.Azure.SignalR.AspNet.Tests/E2ETest/TestHub.cs (hunk 1 above)
Comment (realwanpengli) [reply test/Microsoft.Azure.SignalR.AspNet.Tests/E2ETest/TestHub.cs]: When you call a method in hub, the hub is created by reflection, so the non-static member is created and disposed for every call/request. So I need to kepp it static to use the data across methods.

Comment (vicancy) [reply test/Microsoft.Azure.SignalR.AspNet.Tests/E2ETest/TestHub.cs]: How about using a `ConnectionManager` instead?

**Pull request merged**